src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    const modValue = function(value) {
+      return value === null || value === undefined ? '' : value;
+    };
+
+    for (let i = 0; i < this.length; i++) {
+      result += modValue(this[i]);
```

Comment:
it's better to use if block, because now, if your value is null or undefined you're doing another redundant operation - you're adding an empty string, it's better to avoid it